### PR TITLE
feat(stdlib): added new Window function signature

### PR DESCRIPTION
**Pull request by Bolladeen**
- created new window builtin function that implements new window function signature
  - added to experimental package

### Done checklist
- [ ] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

**Review by scbrickley — APPROVED** [review]
LGTM - Although I haven't spent much time in this code myself. Maybe it's worth it to wait for @nathanielc's review?

File: libflux/flux/benches/benchmarks.rs
Comment (nathanielc) [inline libflux/flux/benches/benchmarks.rs:1082]: @rockstar Thoughts? Should we be trying to keep these benchmarks files up to date when we make changes? Or should we consider them static now?

File: stdlib/universe/window_interval_2.go
Comment (nathanielc) [inline stdlib/universe/window_interval_2.go:29]: This file contains three types:

An OpSpec, ProcedureSpec and a Transformation, all of those are the same as the window_interval.go code. 

I think you can delete all of those from this file and it will still work. All that is needed is that you register the package value as you do on this line here. Then under the hood they become the same function.

File: stdlib/universe/window_test.go
Comment (nathanielc) [inline stdlib/universe/window_test.go:104]: Can you add a Flux acceptance test too? The ones that are written in _test.flux files.

**Review by nathanielc — CHANGES_REQUESTED** [review]
Looks good. I think we can simplify and remove a lot of the copied code, see below.

File: libflux/flux/benches/benchmarks.rs
Comment (rockstar) [reply libflux/flux/benches/benchmarks.rs:1082]: I guess I must've missed when this got added. That's a very big long string that we don't necessarily need in the source. We should change line 23 to be `const LONGER: &'static str = include_str!("path/to/everything.flux")` like we do in other benchmarks to embed it at compile time. I realize it's not strictly part of this PR, but that might be appropriate to do now, since we have to touch it; it'll prevent others from having to make that decision when it needs to be changed next time.

Comment (Bolladeen) [reply libflux/flux/benches/benchmarks.rs:1082]: @rockstar That sounds like a good idea. I'll make that change in this PR.

File: stdlib/universe/window_interval_2.go
Comment (Bolladeen) [reply stdlib/universe/window_interval_2.go:29]: I double checked, but I could be wrong, but I think those three functions are actually different now that we don't include the time, start, and stop columns.

File: stdlib/universe/window_test.go
Comment (Bolladeen) [reply stdlib/universe/window_test.go:104]: Yup, good catch.

**Review by nathanielc — COMMENTED** [review]
This looks great, thanks for making these changes.

I like the name change as well. 

Before we merge I want to have a plan for how we are going to release this and deprecate the other window function. We can chat tomorrow in our team meeting and if that is not enough time we can schedule a working group afterwards.

File: libflux/flux/benches/everything.flux
Comment (rockstar) [inline libflux/flux/benches/everything.flux:1050]: I don't know if we need to add this. We aren't really testing this interface at all. Maybe one day we'll just re-generate `everything.flux` in which case this will get included, but I don't think we should be manually adding things to `everything.flux`.

File: stdlib/universe/universe.flux
Comment (jlapacik) [inline stdlib/universe/universe.flux:85]: This should be `T` so that we can tie the input records to the output records

**Review by jlapacik — COMMENTED** [review]
Just documenting the outcome of our conversation - we decided to require `_start` and `_stop` columns on the input to `window`. So the updated type signature becomes
```
builtin window: (
    <-tables: [{T with _start: time, _stop: time, _time: time}]
    every: duration,
    ?period: duration,
    ?offset: duration,
    ?createEmpty: bool
    ) => [{T with _start: time, _stop: time, _time: time}]
```

File: libflux/flux/benches/everything.flux
Comment (nathanielc) [inline libflux/flux/benches/everything.flux:1050]: nit but can you revert this change?

File: stdlib/experimental/experimental.flux
Comment (nathanielc) [inline stdlib/experimental/experimental.flux:46]: this should specify _start and _stop as per our chat earlier today.

File: stdlib/experimental/window.go
Comment (nathanielc) [inline stdlib/experimental/window.go:9]: this constant isn't really needed now, maybe just use the string `"window"` directly below?

File: stdlib/universe/window.go
Comment (nathanielc) [inline stdlib/universe/window.go:127]: Why was this function added?

**Review by nathanielc — COMMENTED** [review]
Some comments below

File: stdlib/universe/window.go
Comment (Bolladeen) [reply stdlib/universe/window.go:127]: That function was added because we need to export it to the experimental package and go requires an all uppercase function in order to export.

Comment (nathanielc) [reply stdlib/universe/window.go:127]: Can you export the existing function instead of duplicating the function?

File: stdlib/experimental/experimental.flux
Comment (nathanielc) [reply stdlib/experimental/experimental.flux:46]: This is still missing _start and _stop.

**Review by nathanielc — APPROVED** [review]
Looks great! Thanks for getting this all working!

**Pull request merged**
